### DISCUSSION TITLE
Don't stop macro execution on error

### DISCUSSION THREAD

**Discussion by Drllap** [neovim/neovim#24085] [Q&A] [answered]
One thing I have noticed is that macro execution stoppes if there is an error in it, is there a way to change this behavior?
Example:

I have a buffer like:
```
wordone
wordtwo
```
* I move the cursor to the beginning of `wordone`
* do `qq` to start recording
* do `ciwasdf` to change the first ord to `asdf`. Exit insert mode
* press `l` "on accident". This is an error because I'm at the end of the line and I'm probably not going to notice it because I have the bell turned off.
*  do `jciwfle` to change `wordtwo` to `fle`. Exit insert mode
* stop recording macro with `q`
* go the the next occurrence of `wordone`.
* execute macro but only changes `wordone` to `asdf` but nothing happens to `wordtwo`

I would like the macros to continue and also change `wordtwo` even if there is a slite mistake in int.
Any suggestions?

**Comment by zeertzjq** [comment, marked as answer]
It's possible to achieve this using `:silent! normal! @q`

**Reply by nyngwang** [reply]
might be related: Do we have a Lua API for recording? (I was imaging some Lua API that might provide an option to keep execution after an error has happened)

**Reply by zeertzjq** [reply]
No

**Reply by Drllap** [reply]
cool. Is it possible to change the default behavior of executing macros with something like this?

**Reply by vigoux** [reply]
Probably not, though you can create a mapping that mimics that.
The only way I can think of it is through lua:
```lua
local registers = { '0', '1', '2', '3', '4', }
local map = '<Leader>@'

for _, reg in ipairs(registers) do
  local rhs = '<CMD>silent! normal! @' .. reg .. '<CR>'
  vim.api.nvim_set_keymap('n', map .. reg, rhs, {})
end
```

It works like so:
- Record a macro as usual
- Replay the macro using `<Leader>@{register}`

Obviously this only works for registers from 1 to 4 for now, but this is easily extensible by adding elements to the registers table.
